JobIntegrationTest: Increase the restConfig QPS and Burst parameters

tenzen-y opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
As we discussed in the following PR, we decided to increase restConfig QPS and Burst to avoid flaky testing due to client rate-limiting errors.

https://github.com/kubernetes/kubernetes/pull/127078

#### Which issue(s) this PR fixes:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #

#### Special notes for your reviewer:

#### Does this PR introduce a user-facing change?
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

tenzen-y committed:
JobIntegrationTest: Increase the restConfig QPS and Burst parameters

Signed-off-by: Yuki Iwai <yuki.iwai.tz@gmail.com>

tenzen-y commented:
/assign @mimowo @alculquicondor

mimowo reviewed (commented):
LGTM, I would like to also bump it for `BenchmarkLargeIndexedJob` and `BenchmarkLargeFailureHandling` (to align the values). 

I added it for `BenchmarkLargeFailureHandling`, but this was just copy-paste from `BenchmarkLargeIndexedJob`. WDYT @alculquicondor ?

mimowo reviewed (commented):
/lgtm

tenzen-y commented:
> LGTM, I would like to also bump it for `BenchmarkLargeIndexedJob` and `BenchmarkLargeFailureHandling` (to align the values).
> 
> I added it for `BenchmarkLargeFailureHandling`, but this was just copy-paste from `BenchmarkLargeIndexedJob`. WDYT @alculquicondor ?

That is a good point. I updated QPS and Burst for both testing now. If @alculquicondor has any concerns, I can revert it.

alculquicondor commented on test/integration/job/job_test.go:
```diff
@@ -3304,8 +3307,8 @@ func TestElasticIndexedJob(t *testing.T) {
 	closeFn, restConfig, clientSet, ns := setup(b, "indexed")
-	restConfig.QPS = 100
-	restConfig.Burst = 100
+	restConfig.QPS = restConfigQPS
```
such a high QPS makes sense for integration, as we are trying to measure correctness.

But in a benchmark, we want to make sure that the Job controller is fast enough when the QPS is limited.

tenzen-y replied:
That makes sense.
I reverted those, and added comments the reason.
PTAL.

mimowo replied:
Makes sense, thanks for the suggestion and the update.

alculquicondor reviewed (commented):
/lgtm
/approve

mimowo commented:
LGTM
